src/createServer.js

```diff
@@ -1,12 +1,159 @@
+/* eslint-disable no-console */
 'use strict';
 
-// const express = require('express');
+const express = require('express');
 
 function createServer() {
-  // Use express to create a server
-  // Add a routes to the server
-  // Return the server (express app)
-}
+  const app = express();
+
+  const users = [];
+  let userId = 0;
+
+  app.use(express.json());
+
+  app.get('/users', (req, res) => {
+    res.send(users);
+  });
+
+  app.get('/users/:id', (req, res) => {
+    const wantedUser = users.find(user => user.id === +req.params.id);
+
+    if (!wantedUser) {
+      res.sendStatus(404);
+    };
+
+    res.json(wantedUser);
+  });
+
+  app.post('/users', (req, res) => {
+    const newUser = {
+      id: ++userId,
+      name: req.body.name,
+    };
+
+    if (!req.body.name || (users.find(user => user.name === req.body.name))) {
+      res.sendStatus(400);
+
+      return;
+    };
+
+    users.push(newUser);
+    res.status(201).json(newUser);
+  });
+
+  app.patch('/users/:id', (req, res) => {
+    const changedUserIndex = users.findIndex(obj => obj.id === +req.params.id);
+
+    if (changedUserIndex !== -1) {
+      users[changedUserIndex].name = req.body.name;
+      res.status(200).json(users[changedUserIndex]);
+
+      return;
+    };
+
+    res.sendStatus(404);
+  });
+
+  app.delete('/users/:id', (req, res) => {
+    const deletedUserIndex = users.findIndex(obj => obj.id === +req.params.id);
+
+    if (deletedUserIndex !== -1) {
+      users.splice(deletedUserIndex, 1);
+      res.sendStatus(204);
+
+      return;
+    };
+
+    res.sendStatus(404);
+  });
+
+  /* expenses */
+  const epxenses = [];
+  let expensesId = 0;
+
+  app.get('/expenses', (req, res) => {
+    if (req.query.category) {
+      const expensByCategory = epxenses
+        .find(expens => expens.category === req.query.category);
+
+      res.status(200).json([expensByCategory]);
```

Comment:
```suggestion
      res.statusCode = 200;
      res.send(expensByCategory);
```

Reply:
There is no actual difference between res.send and res.json but res.json has an extra option: if we want to send non-object data, res.json would convert it to json format.